client/webserver/site/src/js/app.ts

```diff
@@ -188,16 +188,16 @@ export default class Application {
     const notes = State.fetchLocal(State.notificationsLK)
     this.setNotes(notes || [])
     // Connect the websocket and register the notification route.
-    ws.connect(getSocketURI(), this.reconnected)
+    ws.connect(getSocketURI(), this.reloadPage)
     ws.registerRoute(notificationRoute, (note: CoreNote) => {
       this.notify(note)
     })
   }
 
   /*
-   * reconnected is called by the websocket client when a reconnection is made.
+   * reloadPage is called by the websocket client when a reconnection is made.
    */
-  reconnected () {
+  reloadPage () {
     window.location.reload() // This triggers another websocket disconnect/connect (!)
     // a fetchUser() and loadPage(window.history.state.page) might work
```

Comment:
Anything we'd want to change in this PR regarding that ? The way I understand current design, if page isn't refreshed we'd need at least issue `loadmarket` & `loadcandles` over this WS if we want to keep using it, and maybe page refresh helps with something else too.

Reply:
Investigate for possible subsequent update. Minimal time sink.

Reply:
Using `loadPage` would ensure a complete refresh of all displayed data without the added overhead of an actual page load.

Reply:
OK, see if it works quick then.  If there's any breakage or strange behavior, and it looks more nuanced, we can look at it later.

Reply:
> * when I'm on `/register` page and restart **dexc**, it doesn't auto-recover gracefully (I have to refresh page), and there is an error in console I can't quite make sense of - see screenshot below

I get the same error but only when I'm on another page like `/wallets` that redirects to `/register` on load after dexc restart:

![image](https://user-images.githubusercontent.com/9373513/228675465-ab5908e0-5fa7-4664-a120-d30f30ae86e5.png)

The `data` provided to the `RegistrationPage` constructor is undefined.  This seems to be because we're now doing a `loadPage` for a page that if we had done an http GET we would have gotten a redirect to /register (or /login) from the backend.

I don't think it's quite so simple.  Maybe we don't bother with this approach after all.


Also it might be unrelated, but the wallets page is erroring now until I select an asset, instead of defaulting to one of them:

![image](https://user-images.githubusercontent.com/9373513/228675933-b64ebc50-6da6-42a7-835e-6124b0a52ea1.png)

![image](https://user-images.githubusercontent.com/9373513/228676002-ff929cd8-4940-462b-8291-b8ca3d1bff9a.png)

Reply:
Oh hang on, you're skipping the `fetchUser()` that the comment had instructed should precede the `loadPage`.

Reply:
> Oh hang on, you're skipping the fetchUser() that the comment had instructed should precede the loadPage.

Adding `await app().fetchUser()` right before `app().loadPage(window.history.state.page)` doesn't really change anything for me (seeing same error), and it worked fine on `/markets` page without it too.

Reply:
Alright.  I guess we revert the loadPage changes since we have new bugs now.

Reply:
Removed that last commit then: https://github.com/decred/dcrdex/compare/c53a7ab5bce5401efe464855c75c45cef08ef595..e47a6af4f6d3d903415f9a1baafd8c8134733115

Reply:
This seems to work. https://github.com/buck54321/dcrdex/compare/37acf9f...buck54321:dcrdex:reload-timeout-w-loadpage